server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy

```diff
@@ -1371,93 +1369,89 @@ class IntegrationController {
+                          def additionalInfo = [:]
+
+                          additionalInfo.vars = [the_pkg.id, num_removed_tipps]
+
+                          reviewRequestService.raise(
+                            the_pkg,
+                            "TIPPs retired.",
+                            "An update to package ${the_pkg.id} did not contain ${num_removed_tipps} previously existing TIPPs.",
+                            user,
+                            null,
+                            (additionalInfo as JSON).toString(),
+                            RefdataCategory.lookupOrCreate('ReviewRequest.StdDesc', 'TIPPs Retired')
+                          )
                         }
-                        log.debug("Elapsed tipp processing time: ${System.currentTimeMillis() - tipp_upsert_start_time} for ${tippctr} records")
-                      } else {
-                        job_result.result = 'ERROR'
-                        job_result.message = messageService.resolveCode('crossRef.package.error.cancelled', [json.packageHeader.name], locale)
                       }
-                    }
-                  } else if (cancelled) {
+                      log.debug("Found ${num_removed_tipps} TIPPS to delete/retire from the matched package!")
+                      job_result.result = 'OK'
+                      job_result.message = messageService.resolveCode('crossRef.package.success', [json.packageHeader.name, tippctr, existing_tipps.size(), num_removed_tipps], locale)
+
+                      Package.withNewSession {
+                        def pkg_obj = Package.get(the_pkg.id)
+                        if (pkg_obj.status.value != 'Deleted') {
+                          pkg_obj.lastUpdateComment = job_result.message
+                          pkg_obj.save(flush: true)
+                        }
+                      }
+                      log.debug("Elapsed tipp processing time: ${System.currentTimeMillis() - tipp_upsert_start_time} for ${tippctr} records")
+                    } else {
                     job_result.result = 'ERROR'
                     job_result.message = messageService.resolveCode('crossRef.package.error.cancelled', [json.packageHeader.name], locale)
+                    }
                   } else {
                     job_result.result = 'ERROR'
-                    job_result.message = messageService.resolveCode('crossRef.package.error.tipps', [json.packageHeader.name], locale)
-                    log.warn("Not loading tipps - failed validation")
-
-                    if (the_pkg) {
-                      def additionalInfo = [:]
+                    job_result.message = messageService.resolveCode('crossRef.package.error.cancelled', [json.packageHeader.name], locale)
+                  }
 
-                      if (errors.global.size() > 0 || errors.tipps.size() > 0) {
-                        additionalInfo.errorObjects = errors
-                      }
+                  if (the_pkg) {
```

Comment:
This needs to check for `invalidTipps.size() > 0` as well, otherwise there will be a RR concerning invalid TIPPs for every package import.

Reply:
Sorry for reformatting the whole file. This way, the `if ... else `clauses are easier to fold away in IntelliJ.